### PR TITLE
[MS] Client area personal data modals

### PR DESCRIPTION
**Pull request by NicoTuxx**
## Checklist

Before you submit this pull request, please make sure to:

- [ ] Keep changes in the pull request as small as possible
   
- [ ] Ensure the commit history is sanitized
   
- [ ] Give a meaningful title to your PR
- [ ] Describe your changes
   
- [ ] Link any related issue in the description
   
- [ ] Link any dependent pull request in the description

If this PR adds a new feature or fixes a bug:

- [ ] I have added or updated relevant tests

If this PR modifies installation or updates packages:

- [ ] I have checked that installation works on all platforms (Windows, Linux, Mac ...)
    

- [ ] I have updated related documentations (installation commands, fix to specific problems ...)

If this PR is related to GUI:

- [ ] I have checked that GUIs affected by my changes are working properly.

If this PR affects the user:

- [ ] I have created a news fragment
   
- [ ] I have updated the user documentation

### CODE REVIEW

File: client/src/services/translation.ts [hunk 1]
```diff
@@ -65,3 +65,7 @@ export function getInvitationStatusTranslationKey(status: InvitationStatus): Tra
       return 'UsersPage.invitation.status.cancelled';
   }
 }
+
+export function longLocaleCodeToShort(longCode: string): string {
```
Comment (Max-7) [inline client/src/services/translation.ts:69]: To be a bit pedanctic, we technically have the type `Locale` (https://github.com/Scille/megashark-lib/blob/a543a0a0b47bef7b926375b16f29bbdf1b1d5dfc/lib/services/translation.ts#L24), it ensures that the string has the right format.

File: client/src/views/client-area/AuthenticationModal.vue [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+const props = defineProps<{
+  email: string;
+}>();
+
+const newEmailRef = ref('');
+const passwordRef = ref('');
+const error = ref('');
+const newEmailInput = ref();
+
+const newEmailValidator: IValidator = async function (value: string) {
```
Comment (Max-7) [inline client/src/views/client-area/AuthenticationModal.vue:58]: Doesn't that work?
```
const newEmailValidator: IValidator = async function (value: string) {
  const validity = await emailValidator(value);
  if (validity.validity === Validity.Valid && value.trim() === props.email) {
    return { validity: Validity.Invalid, reason: 'clientArea.personalDataPage.modals.authentication.newEmailMustBeDifferent' };
  }
  return validity;
};
```

File: client/src/views/client-area/PersonalDataPage.vue [hunk 3]
```diff
@@ -75,14 +86,91 @@ function createItem(
...
 });
 
-function openAuthenticationModal(): void {}
+async function getPersonalData(): Promise<void> {
+  personalData.value = await BmsAccessInstance.get().getPersonalInformation();
+}
 
-function openPersonalInfoModal(): void {}
+async function openAuthenticationModal(): Promise<void> {
+  const modal = await modalController.create({
```
Comment (Max-7) [inline client/src/views/client-area/PersonalDataPage.vue:97]: Check that personalData is set and exit early if it isn't. `?` should not be used as a substitute for checks xD

File: client/src/views/client-area/PersonalDataPage.vue [hunk 4]
```diff
@@ -75,14 +86,91 @@ function createItem(
...
+  const { data, role } = await modal.onWillDismiss();
+  await modal.dismiss();
+
+  if (role === MsModalResult.Confirm) {
+    await getPersonalData();
+  }
+}
+
+async function openPersonalInfoModal(): Promise<void> {
+  const modal = await modalController.create({
```
Comment (Max-7) [inline client/src/views/client-area/PersonalDataPage.vue:135]: Same here, exit early if `personData.value` is not set.

File: client/src/views/client-area/PersonalDataPage.vue [hunk 5]
```diff
@@ -75,14 +86,91 @@ function createItem(
...
+  const { data, role } = await modal.onWillDismiss();
+  await modal.dismiss();
+
+  if (role === MsModalResult.Confirm) {
+    await getPersonalData();
+  }
+}
+
+async function openProfessionalInfoModal(): Promise<void> {
+  const modal = await modalController.create({
```
Comment (Max-7) [inline client/src/views/client-area/PersonalDataPage.vue:156]: And same thing here.

File: client/src/views/client-area/SecurityModal.vue [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+const props = defineProps<{
+  email: string;
+}>();
+
+const newEmailRef = ref('');
+const passwordRef = ref('');
+const error = ref('');
+const isNewEmailValid = ref(false);
+
+const newEmailValidator: IValidator = async function (value: string) {
```
Comment (Max-7) [inline client/src/views/client-area/SecurityModal.vue:56]: Dupplicate, also present in AuthenticationModal.

File: client/src/views/client-area/SecurityModal.vue [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+  value = value.trim();
+  if (value.length === 0) {
+    isNewEmailValid.value = false;
```
Comment (Max-7) [inline client/src/views/client-area/SecurityModal.vue:59]: Don't need this variable, use `newEmailRef.value.validity`

**Review by Max-7 — CHANGES_REQUESTED** [review]

File: client/src/views/client-area/PersonalDataPage.vue (hunk 5 above)
Comment (NicoTuxx) [reply client/src/views/client-area/PersonalDataPage.vue:156]: done

Comment (NicoTuxx) [reply client/src/views/client-area/PersonalDataPage.vue:156]: done

File: client/src/views/client-area/PersonalDataPage.vue (hunk 3 above)
Comment (NicoTuxx) [reply client/src/views/client-area/PersonalDataPage.vue:97]: done

File: client/src/services/translation.ts (hunk 1 above)
Comment (NicoTuxx) [reply client/src/services/translation.ts:69]: done

File: client/src/services/bms/types.ts [hunk 8]
```diff
@@ -60,6 +63,18 @@ interface PersonalInformationResultData {
   job?: string;
 }
 
+interface UpdatePersonalInformationResultData {
```
Comment (Max-7) [inline client/src/services/bms/types.ts:66]: I don't think you need empty interfaces if there's no data to return. Just updating the enum is enough.

File: client/src/views/home/HomePageHeader.vue [hunk 9]
```diff
@@ -59,7 +59,7 @@
         </ion-button>
         <ion-button
           fill="solid"
-          v-show="!isElectron()"
+          v-show="isElectron()"
```
Comment (Max-7) [inline client/src/views/home/HomePageHeader.vue:62]: Nop, we don't want to show this button on electron while the client area is not ready.

File: client/src/views/client-area/ProfessionalInfoModal.vue [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+    return false;
+  }
+  const data: any = {
+    company: representCompanyRef.value === Answer.Yes ? companyRef.value : null,
+    job: representCompanyRef.value === Answer.Yes ? jobRef.value : null,
+  };
+  const response = await BmsAccessInstance.get().updatePersonalInformation(data);
+
+  if (response.isError) {
+    // manage errors
```
Comment (Max-7) [inline client/src/views/client-area/ProfessionalInfoModal.vue:66]: Missing some error handling.

File: client/src/views/client-area/PersonalInfoModal.vue [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+    return false;
+  }
+  const response = await BmsAccessInstance.get().updatePersonalInformation({
+    firstname: firstnameRef.value,
+    lastname: lastnameRef.value,
+    phone: phoneRef.value,
+  });
+
+  if (response.isError) {
+    // manage errors
```
Comment (Max-7) [inline client/src/views/client-area/PersonalInfoModal.vue:65]: Missing error handling.

**Review by Max-7 — CHANGES_REQUESTED** [review]

File: client/src/views/home/HomePageHeader.vue (hunk 9 above)
Comment (NicoTuxx) [reply client/src/views/home/HomePageHeader.vue:62]: fixed

File: client/src/services/bms/types.ts (hunk 8 above)
Comment (NicoTuxx) [reply client/src/services/bms/types.ts:66]: done

File: client/src/views/client-area/PersonalInfoModal.vue (hunk 11 above)
Comment (NicoTuxx) [reply client/src/views/client-area/PersonalInfoModal.vue:65]: done

**Review by Max-7 — APPROVED** [review]

**Pull request merged**